packages/client/test/rpc/engine/forkchoiceUpdatedV1.spec.ts

```diff
@@ -271,7 +295,93 @@ tape(`${method}: latest block after reorg`, async (t) => {
   ])
 
   expectRes = (res: any) => {
-    t.equal(res.body.result.payloadStatus.latestValidHash, blocks[1].blockHash)
+    t.equal(res.body.error.code, -32602)
+  }
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(`${method}: validate safeBlockHash is part of canonical chain`, async (t) => {
+  const { server, chain } = await setupChain(genesisJSON, 'post-merge', { engine: true })
+
+  const genesis = await chain.getBlock(BigInt(0))
+
+  // Build the payload for the canonical chain
+  const canonical = [genesis]
+
+  for (let i = 0; i < 2; i++) {
+    canonical.push(createBlock(canonical[canonical.length - 1]))
+  }
+
+  // Build an alternative payload
+  const reorg = [genesis]
+  for (let i = 0; i < 2; i++) {
+    reorg.push(createBlock(reorg[reorg.length - 1]))
+  }
+
+  const canonicalPayload = canonical.map(
+    (e) => blockToExecutionPayload(e, BigInt(0)).executionPayload
+  )
+  const reorgPayload = reorg.map((e) => blockToExecutionPayload(e, BigInt(0)).executionPayload)
+
+  await batchBlocks(t, server, canonicalPayload.slice(1))
+  await batchBlocks(t, server, reorgPayload.slice(1))
+
+  // Safe block hash is not in the canonical chain
+  const req = params(method, [
+    {
+      headBlockHash: reorgPayload[2].blockHash,
+      safeBlockHash: canonicalPayload[1].blockHash,
+      finalizedBlockHash: reorgPayload[1].blockHash,
+    },
+  ])
+
+  const expectRes = (res: any) => {
+    t.equal(res.body.error.code, -32602)
+    t.ok(res.body.error.message.includes('Safe'))
+    t.ok(res.body.error.message.includes('canonical'))
+  }
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(`${method}: validate finalizedBlockHash is part of canonical chain`, async (t) => {
```

Comment:
Note: had to copy the previous test, because if `baseRequest` encounters an error, it ends the test suite thus you cannot test anything else afterwards.